### PR TITLE
VR-4572 create dataset implementation

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java [hunk 1]
```diff
@@ -121,18 +134,24 @@ public void createDataset(
...
       UserInfo userInfo = authService.getCurrentLoginUserInfo();
 
       Dataset dataset = getDatasetFromRequest(request, userInfo);
       ModelDBUtils.checkPersonalWorkspace(
-          userInfo, dataset.getWorkspaceType(), dataset.getWorkspaceId(), "dataset");
-      Dataset createdDataset = datasetDAO.createDataset(dataset, userInfo);
+          userInfo, dataset.getWorkspaceType(), dataset.getWorkspaceId(), "repository");
+      Repository repository =
+          repositoryDAO.createRepository(commitDAO, metadataDAO, dataset, userInfo);
+      Dataset createdDataset =
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java:149]: we will have to populate the other parameters of createDataset

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 2]
```diff
@@ -302,10 +312,46 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
...
+          GET_REPOSITORY_COUNT_BY_NAME_PREFIX_HQL,
+          RepositoryEntity.class.getSimpleName(),
+          "repositoryName",
+          repository.getName(),
+          ModelDBConstants.WORKSPACE_ID,
+          workspaceDTO.getWorkspaceId(),
+          workspaceDTO.getWorkspaceType(),
+          LOGGER);
+      repositoryEntity = new RepositoryEntity(repository, workspaceDTO);
+      repositoryEntity.setDeleted(true);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: this logic has changed on master

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 3]
```diff
@@ -314,66 +360,40 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
...
-                authService.getVertaIdFromUserInfo(userInfo),
-                repositoryEntity);
-
-        saveBranch(
-            session,
-            commitEntity.getCommit_hash(),
-            ModelDBConstants.MASTER_BRANCH,
-            repositoryEntity);
-      }
+      repositoryEntity.setDeleted(false);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: please pull in master to update on the repo creation logic

**Review by ravishetye — APPROVED** [review]
this makes sense , but please pull in master into feature since the repo creation logic was updated for a bug fix

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java:149]: What other parameters? Attributes, tags, date created / updated is already put in dataset structure before createRepository call. I don't see what's left.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 2 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: pulled master

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 3 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: pulled master

**Pull request merged**